pyccel/parser/semantic.py

```diff
@@ -740,6 +740,10 @@ def _infer_type(self, expr):
             d_var['memory_handling'] = 'heap'
             return d_var
 
+        elif isinstance(expr, PythonList):
+            d_var['cls_base'       ] = ListClass
+            d_var['memory_handling'] = 'heap'
+            return d_var
```

Comment:
Is this condition necessary? Can you explain why this is not handled by the condition below `elif isinstance(expr, TypedAstNode):`? (Maybe `PythonTuple` caused confusion? The condition `elif isinstance(expr, PythonTuple):` is removed in one of my unfinished PRs as it is no longer necessary)

Reply:
Fixed 7bd019398f4fa6e6b0f7050c9bbc38724fdabd2e